plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/TypeUtils.java

```diff
@@ -142,14 +142,23 @@ private static Object jdbcObjectToTrinoNative(ConnectorSession session, Object j
         }
 
         if (BOOLEAN.equals(type)
-                || TINYINT.equals(type)
-                || SMALLINT.equals(type)
-                || INTEGER.equals(type)
                 || BIGINT.equals(type)
                 || DOUBLE.equals(type)) {
             return jdbcObject;
         }
 
+        if (TINYINT.equals(type)) {
+            return ((Byte) jdbcObject).longValue();
+        }
+
+        if (SMALLINT.equals(type)) {
+            return ((Short) jdbcObject).longValue();
+        }
+
+        if (INTEGER.equals(type)) {
+            return ((Integer) jdbcObject).longValue();
```

Comment:
i would consider writing this as:

```suggestion
            return (long) (int) jdbcObject;
```

-- avoids boxing.